### PR TITLE
Changed the way org commanders validate requests

### PR DESCRIPTION
**Pull request by ryanpbrewster**
Now they provide an `Option[OrganizationFail]` instead of a `Boolean`, so
other methods no longer have to guess at the reason that a request didn't
validate.

@camhashemi

### CODE REVIEW

**Comment by camhashemi** [comment]
Looks good

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationInviteCommander.scala [hunk 1]
```diff
@@ -57,20 +57,26 @@ class OrganizationInviteCommanderImpl @Inject() (db: Database,
...
         val requesterOpt = organizationMembershipRepo.getByOrgIdAndUserId(orgId, requesterId)
         val canRequesterCancelInvites = requesterOpt.exists { _.permissions.contains(INVITE_MEMBERS) }
-        val doInvitesExist = {
+        val doInvitesExist: Boolean = {
           val existingInvites = organizationInviteRepo.getAllByOrganization(request.orgId)
           targetEmails.forall(email => existingInvites.exists(_.emailAddress.contains(email))) &&
             targetUsers.forall(userId => existingInvites.exists(_.userId.contains(userId)))
         }
-        canRequesterCancelInvites && doInvitesExist
+        if (!canRequesterCancelInvites) Some(OrganizationFail.INSUFFICIENT_PERMISSIONS)
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationInviteCommander.scala:77]: need to check for `OrganizationFail.NOT_A_MEMBER`

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationInviteCommander.scala [hunk 2]
```diff
@@ -273,24 +279,24 @@ class OrganizationInviteCommanderImpl @Inject() (db: Database,
...
-      emailInvitesToCancel.foreach(organizationInviteRepo.deactivate)
-      userIdInvitesToCancel.foreach(organizationInviteRepo.deactivate)
-
-      Right(OrganizationInviteCancelResponse(
-        request,
-        cancelledEmails = emailInvitesToCancel.map(_.emailAddress.get),
-        cancelledUserIds = userIdInvitesToCancel.map(_.userId.get))
-      )
+    getValidationError(request) match {
+      case Some(fail) => Left(OrganizationFail.INSUFFICIENT_PERMISSIONS)
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationInviteCommander.scala:283]: should return `Left(fail)`

**Comment by camhashemi** [comment]
I'll take care of those two minor things

**Comment by ryanpbrewster** [comment]
Cool, thanks.

**Pull request merged**